### DISCUSSION TITLE
FEA field quantities

### DISCUSSION THREAD

**Discussion by SebGue** [Eomys/pyleecan#441] [Ideas]
Hello,

I was wondering if it is needed to save all of the field quantities (i.e. B, H and mu) of the FEA calculations. If we e.g. would only store the magnetic field B all of the other 2 could be reconstructed.
So this is basically a question about the tradeoff between speed (for field reconstruction) vs. storage space (especially on the hard drive, which may also affect loading/saving speed).

Maybe the decision which fields to keep should be set by the user (default all fields)?!

Do you use all fields? Whats your general opinion?

By the way this could also been extended to the magnetic vector field.

Best regards, Sebastian

**Comment by EmileDvs** [comment]
Hello @SebGue,

I agree on the fact that the user should know which fields are relevant to keep. 

I also agree on the trade off part. As far as I know, postprocessing all quantities (A, B, H and mu) using lua script is very fast, and storing becomes problematic only for huge models and many time steps. 

Assuming you can recalculate all quantities, the best option to reduce storage is to only store magnetic vector potential along z axis (Az). From this you get flux density and flux lines (B=rot(A)), and then H and mu from BH curve. The question is: how long does it takes and how accurate is it to get B, H, mu from Az and BH curve compared with using lua script ? If you already developed such methods, you can also optimize the lua script only extract data requested by the user (at the moment it extracts all the data from the .ans file).

We also intend to improve .h5 save/load functions since they can be quite slow for huge Meshsolution objects. A speed up of save/load functions might also be a solution to your issue.

Best regards,
Emile

**Comment by SebGue** [comment]
Hello @EmileDvs ,

storing space is my main concern for now since I don't use h5. But with the upcoming compressed json that maybe not an issue anymore. I only got to check the code again so it can be merged. Speed comparison between h5 and compressed json will also be interesting.

For the field quantities I'm not quite sure. The calculation of B=f(Az), and mu=f(B,H) may be fast but calculation of H should be a lot slower due to the nonlinear BH curve. Further the calculation of H always requires the knowledge of the used BH curve. So I don't know if H is even reconstructable without additional affort.

The accuracy of 'lua and Az' B should be the same as far as I know. I had some matlab script where I was reading the ans files (they store Az only) directly and calculated B from it. Maybe I will find it. So I could translate it to have a comparison. Anyway ans files are easy to understand, so writing a function from scratch isn't too hard.

Best regards, Sebastian
